src/main/java/com/aws/greengrass/shadowmanager/configuration/RateLimitsConfiguration.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.shadowmanager.configuration;
+
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.shadowmanager.util.Validator;
+import com.aws.greengrass.util.Coerce;
+import lombok.Getter;
+
+import static com.aws.greengrass.shadowmanager.model.Constants.CONFIGURATION_MAX_LOCAL_REQUESTS_RATE_PER_THING_TOPIC;
+import static com.aws.greengrass.shadowmanager.model.Constants.CONFIGURATION_MAX_OUTBOUND_UPDATES_PS_TOPIC;
+import static com.aws.greengrass.shadowmanager.model.Constants.CONFIGURATION_MAX_TOTAL_LOCAL_REQUESTS_RATE;
+import static com.aws.greengrass.shadowmanager.model.Constants.CONFIGURATION_RATE_LIMITS_TOPIC;
+import static com.aws.greengrass.shadowmanager.model.Constants.DEFAULT_LOCAL_REQUESTS_RATE;
+import static com.aws.greengrass.shadowmanager.model.Constants.DEFAULT_MAX_OUTBOUND_SYNC_UPDATES_PS;
+import static com.aws.greengrass.shadowmanager.model.Constants.DEFAULT_TOTAL_LOCAL_REQUESTS_RATE;
+
+public final class RateLimitsConfiguration {
+    @Getter
+    private final int maxTotalLocalRequestRate;
+    @Getter
+    private final int maxLocalRequestRatePerThing;
+    @Getter
+    private final int maxOutboundUpdatesPerSecond;
+
+    private static int getMaxTotalLocalRequestRateFromTopics(Topics rateLimitsTopics) {
+        int maxTotalLocalRequestRate = Coerce.toInt(rateLimitsTopics
+                .lookup(CONFIGURATION_MAX_TOTAL_LOCAL_REQUESTS_RATE)
+                .dflt(DEFAULT_TOTAL_LOCAL_REQUESTS_RATE));
+        Validator.validateTotalLocalRequestRate(maxTotalLocalRequestRate);
+        return maxTotalLocalRequestRate;
+    }
+
+    private static int getMaxLocalRequestRatePerThingFromTopics(Topics rateLimitsTopics) {
+        int maxLocalRequestRatePerThing = Coerce.toInt(rateLimitsTopics
+                .lookup(CONFIGURATION_MAX_LOCAL_REQUESTS_RATE_PER_THING_TOPIC)
```

Comment:
Prefer to use find or default rather than dflt. Dflt sets an initial value, but that value wont act like the default that you're expecting. Ie. It will not necessarily be used when the config value is missing.

Reply:
Updated.